widgetastic_manageiq/__init__.py

```diff
@@ -4051,3 +4051,461 @@ def data_for_timestamp(self, timestamp):
+                    partial_match=partial_match)
+                if handle_alert and self.browser.alert_present:
+                    # TODO: Investigate why alert_present is False
+                    alert = self.browser.get_alert()
+                    if 'source datastores exceeds the available space' in alert.text:
+                        self.logger.info("{} cannot be mapped to target {} due to lack of space on"
+                            " target".format(item, target))
+                else:
+                    self.add_mapping.click()
+
+
+class InfraMappingWizardNetworksViw(InfraMappingCommonButtonsSet1, InfraMappingCommonButtonsSet2):
+    source_networks = MultiSelectBox('source_networks')
+    target_networks = MultiSelectBox('target_networks')
+    next_btn = Button("Create")  # overriding, since 'Next' is called 'Create' in this form
+    cluster_selector = BootstrapSelect(locator='.//div[contains(@class,"cluster_select_select")]')
+
+    @property
+    def is_displayed(self):
+        return self.source_networks.item_count > 0 or self.target_networks.item_count > 0
+
+    def fill(self, values, partial_match=False):
+        """Use _add_cluster_mapping to add all mappings specified in values.
+
+        Args:
+        values:
+            format as following:
+                {
+                    'mapping1':
+                    {
+                        'sources':['item1','item2'],
+                        'target': 'item_target1'
+                    },
+                    'mapping2':
+                    {
+                        'sources':['item3','item4'],
+                        'target': 'item_target2'
+                    },
+                    ...
+                }
+        """
+        try:
+            for mapping in values:
+                self._add_network_mapping(values[mapping]['sources'],
+                    values[mapping]['target'], partial_match=partial_match)
+            return True
+        except Exception as e:
+            self.logger.debug("Following Exception occured while trying to fill:{}".format(e))
+            return False
+
+    def _add_network_mapping(self, sources=[], target=None, partial_match=False):
+        """Add mapping from one or more sources to one target
+.
+        Args:
+            sources: list of source items
+            target: str containing target item
+        """
+        if isinstance(sources, list) and sources and target:
+            for item in sources:
+                self.source_networks.select_by_text(item, partial_match)
+                self.target_networks.select_by_text(target.split("(")[0],
+                    partial_match=partial_match)
+                self.add_mapping.click()
+
+
+class InfraMappingWizardResultsViw(InfraMappingCommonButtonsSet1):
+    next_btn = Button("Close")
+    continue_to_plan_wizard_btn = Button("Continue to the plan wizard")
+
+
+class InfraMappingWizard(Widget):
```

Comment:
**REQUIRED**
This should be a view.

Reply:
@mshriver I believe we discussed this to be a widget

Reply:
Nope, it should be a view.

Reply:
@kedark3 was there a reason for using `Widget` instead of `View` here?  `View` definitely makes more sense as its a modal that's a collection of widgets.